sdk/storage/storage-blob/test/blobclient.spec.ts

```diff
@@ -536,7 +536,7 @@ describe("BlobClient", () => {
       containerName,
       "Container name is not the same as the one provided."
     );
-    assert.equal(newClient.blobName, blobName, "Blob name is not the same as the one provided.");
+    assert.equal(newClient.name, blobName, "Blob name is not the same as the one provided.");
```

Comment:
Minor: Might be worth to be noted. Through this specific example, it looks like it's not that intuitive for user to guess what's the name for. Possibly because blobClient seems not equal to blob, and also user might give a casual name for client, so we may lose the concept binding, and hard to guess what's name for a newClient is.

Reply:
Completely agree with you! 
Should we consider `blobName` over `name`?

cc - @jeremymeng, @bterlson

Reply:
Wouldn't the same apply for queues as well?

Since I saw elsewhere that for naming, we are trying to be consistent with other languages and .Net is one of the languages that is ahead in making the changes that were discussed, I looked at what they were doing.

.Net is using [Name for Queues](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Queues/src/QueueClient.cs#L100) and [Name for Blobs](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs#L110) too

Reply:
Hi, @ramya-rao-a 
Yes, same comments applies to other changes. And agree that alignment might be much important. This comment is only my personal 2 cents. Personally, I prefer to blobName, queueName, dirPath and filePath, as in this example, it looks like much intuitive?

The PR itself is good to me.

Reply:
Thanks @jiacfan 
@XiaoningLiu Do you have any inputs here?

Reply:
`name` is consistent with other languages. It's unlikely that the other languages would adapt to `blobName` at this point. 

Let's merge this?

cc - @tg-msft